components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java

```diff
@@ -179,16 +229,68 @@ public List<UserSession> getSessionsByUserId(String userId) throws SessionManage
         if (log.isDebugEnabled()) {
             log.debug("Retrieving all the active sessions of user: " + userId + ".");
         }
-        return getActiveSessionList(getSessionIdListByUserId(userId));
+
+        List<UserSession> userSessions;
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        // First check whether a federated association exists for the userId.
+        try {
+            int tenantId = getTenantId(tenantDomain);
+            String fedAssociatedUserId = getUserIdFromFederatedMapping(tenantId, userId);
+
+            if (StringUtils.isNotEmpty(fedAssociatedUserId)) {
+                userSessions = getActiveSessionList(
+                        fedAssociatedUserId, getSessionIdListByUserId(fedAssociatedUserId));
+            } else {
+                userSessions = getActiveSessionList(userId, getSessionIdListByUserId(userId));
```

Comment:
Requires to merge these two results to accommodate linking local users to another user.
Ex: JIT provisioned user entering a password and allowing it to login with local userId.

Reply:
Addressed with 7056d26ef22a1c38205b79e2a0143791c98761b8